### ISSUE TITLE
$1 handled different than variable in SC1087 Use braces when expanding arrays

### ISSUE THREAD

**Issue by ChillerDragon** [koalaman/shellcheck#2910]
#### For bugs
- Rule Id: SC1087
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC1087)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

export FOO="$1[@]" # no error
export FOO="$FOO[@]" # SC1087 (error): Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).
```

#### Here's what shellcheck currently says:

no error

#### Here's what I wanted or expected to see:

SC1087 (error): Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).
